src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1934,9 +1934,36 @@ private class OffsetAndRule {
 
 
 #if PLATFORM_UNIX
-        static public TimeZoneInfo FindSystemTimeZoneById(string id) {
-            // UNIXTODO
-            throw new NotImplementedException();
+        static public TimeZoneInfo FindSystemTimeZoneById(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException("id");
+            }
+
+            // Special case for Utc since there is code that expects only 1 instance of the UTC time zone info to be created.
+            // see CachedData.GetCorrespondingKind()
+            if (String.Compare(id, c_utcId, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return TimeZoneInfo.Utc;
+            }
+
+            // UNIXTODO: use CachedData
+
+            string timeFilePath = Path.Combine(c_timeZoneDirectory, id);
+            if (!File.Exists(timeFilePath))
+            {
+                throw new TimeZoneNotFoundException(Environment.GetResourceString("TimeZoneNotFound_MissingRegistryData", id));
```

Comment:
Based on the name of this resource key, I'm wondering if we may need to use a different error message, or at least verify that the error message in use doesn't say anything about the Windows registry.